src/ui/ui_settings/ui_settings_mixin.js

```diff
@@ -9,9 +10,35 @@ export function uiSettingsMixin(kbnServer, server, config) {
     return;
   }
 
-  const uiSettings = new UiSettings(server, status);
-  server.decorate('server', 'uiSettings', () => uiSettings);
+  // Passed to the UiSettingsService.
+  // UiSettingsService calls the function before trying to read data from
+  // elasticsearch, giving us a chance to prevent it from happening.
+  //
+  // If the ui settings status isn't green we shouldn't be attempting to get
+  // user settings, since we can't be sure that all the necessary conditions
+  // (e.g. elasticsearch being available) are met.
+  const readUiSettingsInterceptor = () => {
```

Comment:
I'm theoretically a huge fan of never using `undefined` to represent nothing and using `null` instead (reserving `undefined` to mean `uninitialized` or something).

In practice though I find that opinion pretty controversial and think it's somewhat impractical to require. Also, if we requiring interceptors to return `null` seems like an unexpected API, based on my experience with such things... In general I treat `null` and `undefined` as equivalent, which is why I use `== null` a lot.